src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -284,7 +284,11 @@ bool Condition::Meets(ConditionSourceInfo& sourceInfo) const
         }
         case CONDITION_NEAR_CREATURE:
         {
-            condMeets = GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, bool(!ConditionValue3)) ? true : false;
+            Creature* crea = GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, bool(!ConditionValue3));
+            if (!crea || (ConditionValue3 && crea->getDeathState() == DEAD))
```

Comment:
Also this isn't going to work if there's, say, a despawned corpse at 15yd and a living creature at 20yd.